src/realm.js

```diff
@@ -657,6 +657,33 @@ export class Realm {
     return new Constructor(this, types, values, args, buildNode, { kind, intrinsicName });
   }
 
+  makeInstrinsicObject(ob: ObjectValue, intrinsicName: string, visited?: Set<ObjectValue> = new Set()): ObjectValue {
+    ob.intrinsicName = intrinsicName;
+    ob.intrinsicNameGenerated = true;
+    for (let [key, binding] of ob.properties) {
+      if (binding === undefined || binding.descriptor === undefined) continue; // deleted
+      invariant(binding.descriptor !== undefined);
+      let value = binding.descriptor.value;
+      ThrowIfMightHaveBeenDeleted(value);
+      if (value === undefined) {
+        AbstractValue.reportIntrospectionError(ob, key);
+        throw new FatalError();
+      }
+      let propertyName = `${intrinsicName}.${key}`;
+      if (value instanceof ObjectValue) {
+        if (!visited.has(value)) {
+          visited.add(value);
+          this.makeInstrinsicObject(value, propertyName, visited);
+        }
+      } else if (value instanceof AbstractValue && !value.isIntrinsic()) {
+        value.intrinsicName = propertyName;
+        value.args = [ob];
+        value._buildNode = ([node]) => t.memberExpression(node, t.identifier(key));
```

Comment:
This follows the pattern established by someone else (you?) in Realm.rebuildObjectProperty. The serializer expects things this way.

The basic question is pertinent, however. In setting up intrinsic objects this way, we implicitly require intrinsic objects to have pure properties. I suspect that we can't get by without this requirement, though.